operator/v1/types_ingress.go

```diff
@@ -282,6 +287,178 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the schema for securing an IngressController.
+type SecuritySpec struct {
+	// profile defines the schema for a security profile.
+	Profile SecurityProfileSpec `json:"profile"`
+}
+
+// SecurityProfileSpec defines the schema for a security profile.
+// +union
+type SecurityProfileSpec struct {
+	// type is one of type OldSecurityProfile, IntermediateSecurityProfile,
+	// ModernSecurityProfile or CustomSecurityProfile. OldSecurityProfile,
+	// IntermediateSecurityProfile and ModernSecurityProfile profiles map to
+	// security compatiibility configurations defined by:
```

Comment:
Should reference the actual constant values and a description of what each means, see the EndpointPublishingStrategyType godoc for a reference

Reply:
I'll update.